### PR TITLE
Torch mode implementation

### PR DESCRIPTION
**Pull request by jqn**
@ekreutz could you please look into merging this pull request. This is the torch mode implementation only, I've removed everything else. The original author of this pull request doesn't seem to be active and it would be really useful to have this functionality included in this package.

### CODE REVIEW

File: android/src/main/java/com/ekreutz/barcodescanner/ui/BarcodeScannerView.java [hunk 1]
```diff
@@ -223,6 +233,19 @@ public boolean setFocusMode(int focusMode) {
...
+     * Sets torch mode.
+     * Possible values: 0 = off, 1 = torch (always on)
+     * @param torchMode
+     */
+    public boolean setTorchMode(int torchMode) {
+        if (torchMode < 0 || torchMode > 1) {
+            torchMode = 0;
+        }
+
+        return mCameraSource != null && mCameraSource.setFlashMode(RELEVANT_FLASH_MODES[torchMode]);
```
Comment (avshenuk) [inline android/src/main/java/com/ekreutz/barcodescanner/ui/BarcodeScannerView.java]: On some devices FLASH_MODE_TORCH is not supported and the best way is to specify FLASH_MODE_ON.
if (mCameraSource != null) {
            if (torchMode == 0) {
                return mCameraSource.setFlashMode(Camera.Parameters.FLASH_MODE_OFF);
            }
            return mCameraSource.setFlashMode(Camera.Parameters.FLASH_MODE_TORCH)
                    || mCameraSource.setFlashMode(Camera.Parameters.FLASH_MODE_ON);
        }

Comment (jqn) [reply android/src/main/java/com/ekreutz/barcodescanner/ui/BarcodeScannerView.java]: Ok, I'll update my pull request

**Comment by jqn** [comment]
@ekreutz  correct me if I'm wrong but if FLASH_MODE_TORCH isn't available there is already a logic handling this case in CameraSource.java line 840.

```
if (mFlashMode != null) {
            if (parameters.getSupportedFlashModes() != null) {
                if (parameters.getSupportedFlashModes().contains(
                        mFlashMode)) {
                    parameters.setFlashMode(mFlashMode);
                } else {
                    Log.i(TAG, "Camera flash mode: " + mFlashMode + " is not supported on this device.");
                }
            }
        }
```

and setting the flash mode to on doesn't act like torch mode anyways since it is not continuously enabled. For that I believe the we would need to add a flashmode property.

**Comment by TheCodeTalker** [comment]
hey this pr what implmented is working

**Comment by jqn** [comment]
it is working just not getting a lot of traction with the repo owner to merge it.

**Comment by TheCodeTalker** [comment]
@ekreutz FLASH not working i have tested with android phone

**Pull request merged**